src/main/scala/com/fulcrumgenomics/commons/io/CommandLineTool.scala

```diff
@@ -0,0 +1,205 @@
+    * @param args a variable list of arguments to pass to the script
+    * @throws Exception when we are unable to execute to this script on the classpath with the given arguments.
+    */
+  def execIfAvailable(scriptResource: String, args: String*): Unit =
+    try { if (Available) exec(scriptResource, args:_*) }
+    catch { case e: Exception => throw new Exception(s"Cannot execute script from $scriptResource with args:$args") }
+
+  /** Executes a script stored at a Path if the this executable is available.
+    *
+    * @param script Path of the script to be run
+    * @param args a variable list of arguments to pass to the script
+    * @throws Exception when we are unable to execute to this script on the classpath with the given arguments.
+    */
+  def execIfAvailable(script: Path, args: String*): Unit = execIfAvailable(script.toString, args: _*)
+
+  /** Executes a script from the classpath if this executable is available, raise an exception otherwise.
+    *
+    * @throws ToolException when the exit code from the called process is not zero.
+    * @throws Exception when we are unable to execute to this script on the classpath with the given arguments.
+    * @param scriptResource the name of the script resource on the classpath
+    * @param args a variable list of arguments to pass to the script
+    * @return Unit as this code is side-effecting
+    */
+  def exec(scriptResource: String, args: String*): Unit = exec(writeResourceToTempFile(scriptResource), args:_*)
+
+  /** Execute from a script stored at a Path if this executable is available, raise an exception otherwise.
+    *
+    * @throws ToolException when the exit code from the called process is not zero.
+    * @throws Exception when we are unable to execute to this script with the given arguments.
+    * @param script Path of the script to be run
+    * @param args a variable list of arguments to pass to the script
+    * @return Unit as this code is side-effecting
+    */
+  def exec(script: Path, args: String*): Unit = try {
+    val command = Executable +: script.toAbsolutePath.toString +: args
+    val process = new ProcessBuilder(command:_*).redirectErrorStream(false).start()
+    val pipe1   = Io.pipeStream(process.getErrorStream, logger.info)
+    val pipe2   = Io.pipeStream(process.getInputStream, logger.debug)
+    val retval  = process.waitFor()
+    pipe1.close()
+    pipe2.close()
+
+    if (retval != 0) throw ToolException(retval)
+  } catch {case e: Exception => throw new Exception(s"Cannot execute script from path $script with args: $args") }
+
+  /** Extract a resource from the classpath and writes it to a temp file on disk.
+    *
+    * @param resource a given name on the classpath
+    * @return path to the temporary file
+    * */
+  private def writeResourceToTempFile(resource: String): Path = {
+    val lines = Io.readLinesFromResource(resource).toSeq
+    val path  = Io.makeTempFile("script.", suffix = Suffix)
+    path.toFile.deleteOnExit()
```

Comment:
I don't think you need to specify `deleteOnExit()`, `Io.makeTempFile()` should take care of that for you

Reply:
@mjhipp no sure if it's already taken care of?

Here's the docs for createTempFile:
 
https://docs.oracle.com/javase/7/docs/api/java/nio/file/Files.html#createTempFile(java.nio.file.Path,%20java.lang.String,%20java.lang.String,%20java.nio.file.attribute.FileAttribute... 

> Where used as a work files, the resulting file may be opened using the DELETE_ON_CLOSE option so that the file is deleted when the appropriate close method is invoked. Alternatively, a shutdown-hook, or the File.deleteOnExit() mechanism may be used to delete the file automatically.